state/README.md

```diff
@@ -0,0 +1,148 @@
+# Using State Persistence in an Extension
+
+This example shows how to save and restore data save in persistent state database
+in a JupyterLab extension.
+
+The core token required for handling the state database (DB) is
+`IStateDB` ([documentation](https://jupyterlab.github.io/jupyterlab/coreutils/modules/istatedb.html)). To use it,
+you first need to install its npm package:
+
+```bash
+jlpm add @jupyterlab/coreutils
+```
+
+Once this is done. You can import the interface in your code.
+
+```ts
+// src/index.ts#L8-L8
+
+import { IStateDB } from '@jupyterlab/coreutils';
+```
+
+To see how we can access the state, let's have a look at
+`src/index.ts`.
+
+```ts
+// src/index.ts#L17-L56
+
+const extension: JupyterFrontEndPlugin<void> = {
+  id: PLUGIN_ID,
+  autoStart: true,
+  requires: [IStateDB],
+  activate: (app: JupyterFrontEnd, state: IStateDB) => {
+    const choices = ['one', 'two', 'three'];
+    let choice = choices[0];
+
+    app.restored
+      // Get the state object
+      .then(() => state.fetch(PLUGIN_ID))
+      .then(value => {
+        // Get the choice attribute
+        if (value) {
+          choice = (value as ReadonlyJSONObject)['choice'] as string;
+        }
+
+        // Ask the user to pick a choice with `choice` as default
+        return InputDialog.getItem({
+          title: 'Pick a choice',
+          items: choices,
+          current: Math.max(0, choices.indexOf(choice))
+        });
+      })
+      .then(result => {
+        // If the user click on the accept button of the dialog
+        if (result.button.accept) {
+          // Get the user choice
+          choice = result.value;
+          // Save the choice in the state database
+          return state.save(PLUGIN_ID, { choice });
+        }
+      })
+      .catch(reason => {
+        console.error(
+          `Something went wrong when reading the state for ${PLUGIN_ID}.\n${reason}`
+        );
+      });
+  }
+};
+```
+
+The `IStateDB` is passed to the `activate` function as an
+argument (variable `state`) in addition to the JupyterLab application
+(variable `app`). You request that dependency with the property
+`requires: [IStateDB],`. It lists the additional arguments
+you want to inject into the `activate` function in the `JupyterFontEndPlugin`.
+
+First, the extension waits for the application to be restored before
+loading the state data for your plugin:
+
+<!-- prettier-ignore-start -->
+```ts
+// src/index.ts#L25-L27
+
+app.restored
+  // Get the state object
+  .then(() => state.fetch(PLUGIN_ID))
+```
+<!-- prettier-ignore-end -->
+
+The data are loaded as a `ReadonlyJSONValue` object. So all stored data must be
```

Comment:
I know it's usually debated, but I think the JupyterLab comments and docs use the singular for data:

```suggestion
The data is loaded as a `ReadonlyJSONValue` object. So all stored data must be
```